File: TESS_PythonAPI_demo/新闻案例代码/2车辆速度引导/TESS_API_EXAMPLE.py
# -*- coding: utf-8 -*-

import os
from pathlib import Path

from Tessng import *
from DockWidget import *


class TESS_API_EXAMPLE(QMainWindow):
    def __init__(self, parent = None):
        super(TESS_API_EXAMPLE, self).__init__(parent)
        self.ui = Ui_TESS_API_EXAMPLEClass()
        self.ui.setupUi(self)
        self.createConnect()

    def createConnect(self):
        self.ui.btnOpenNet.clicked.connect(self.openNet)
        self.ui.btnStartSimu.clicked.connect(self.startSimu)
        self.ui.btnPauseSimu.clicked.connect(self.pauseSimu)
        self.ui.btnStopSimu.clicked.connect(self.stopSimu)

    def openNet(self):
        iface = tngIFace()
        if not iface:
            return
        if iface.simuInterface().isRunning():
            QMessageBox.warning(None, "提示信息", "请先停止仿真，再打开路网")
            return
        custSuffix = "TESSNG Files (*.tess);;TESSNG Files (*.backup)"
        dbDir = os.fspath(Path(__file__).resolve().parent / "Data")
        selectedFilter = "TESSNG Files (*.tess)"
        options = QFileDialog.Options(0)
        netFilePath, filtr = QFileDialog.getOpenFileName(self, "打开文件", dbDir, custSuffix, selectedFilter, options)
        if netFilePath:
            iface.netInterface().openNetFle(netFilePath)

    def startSimu(self):
        iface = tngIFace()
        if not iface:
            return
        if not iface.simuInterface().isRunning() or iface.simuInterface().isPausing():
            iface.simuInterface().startSimu()

    def pauseSimu(self):
        iface = tngIFace()
        if not iface:
            return
        if iface.simuInterface().isRunning():
            iface.simuInterface().pauseSimu()

    def stopSimu(self):
        iface = tngIFace()
        if not iface:
            return
        if iface.simuInterface().isRunning():
            iface.simuInterface().stopSimu()

    def showRunInfo(self, runInfo):
        self.ui.txtMessage.clear()
        self.ui.txtMessage.setText(runInfo)

    def isOk(self):
        QMessageBox.information(None, "提示信息", "is ok!")

File: TESS_CppAPI_EXAMPLE/TESS_API_EXAMPLE.cpp
#include "TESS_API_EXAMPLE.h"

#include <QPushButton>
#include <QFileDialog>
#include <QMessageBox>

#include "tessinterface.h"
#include "netinterface.h"
#include "simuinterface.h"
#include "guiinterface.h"

TESS_API_EXAMPLE::TESS_API_EXAMPLE(QWidget *parent): QMainWindow(parent){
	ui.setupUi(this);

	QObject::connect(ui.btnOpenNet, &QPushButton::clicked, [this]() {
		openNet();
	});

	QObject::connect(ui.btnStartSimu, &QPushButton::clicked, [this]() {
		startSimu();
	});

	QObject::connect(ui.btnPauseSimu, &QPushButton::clicked, [this]() {
		pauseSimu();
	});

	QObject::connect(ui.btnStopSimu, &QPushButton::clicked, [this]() {
		stopSimu();
	});

}

void TESS_API_EXAMPLE::showRunInfo(const QString &runInfo) {
	ui.txtMessage->clear();
	ui.txtMessage->setText(runInfo);
}

//打开路网
void TESS_API_EXAMPLE::openNet() {
	//if (gpTessInterface->simuInterface()->isRunning()) {
	//	QMessageBox::warning(nullptr, QString(), QString("请先停止仿真，再打开路网"));
	//	return;
	//}

	QString	custSuffix = QString("TESSNG Files (*.tess);;TESSNG Files (*.backup)");
	QString dbDir = QApplication::applicationDirPath() + "/Data";
	QString selectedFilter = QString("TESSNG Files (*.tess)");

	const QFileDialog::Options options = QFlag(0);

	QString netFilePath = QFileDialog::getOpenFileName(this,
		QString("打开文件"),
		dbDir,
		custSuffix,
		&selectedFilter,
		options);

	if (!netFilePath.isEmpty()){
		gpTessInterface->netInterface()->openNetFle(netFilePath);
	}
}

//启动仿真
void TESS_API_EXAMPLE::startSimu() {
	if (!gpTessInterface->simuInterface()->isRunning() || gpTessInterface->simuInterface()->isPausing()) {
		gpTessInterface->simuInterface()->startSimu();
	}
}

//暂停仿真
void TESS_API_EXAMPLE::pauseSimu() {
	if (gpTessInterface->simuInterface()->isRunning()) {
		gpTessInterface->simuInterface()->pauseSimu();
	}
}

//停止仿真
void TESS_API_EXAMPLE::stopSimu() {
	if (gpTessInterface->simuInterface()->isRunning()) {
		gpTessInterface->simuInterface()->stopSimu();
	}
}